cpp-client/deephaven/client/include/public/deephaven/client/utility/table_maker.h

```diff
@@ -17,6 +17,8 @@
 #include <arrow/util/key_value_metadata.h>
 #include "deephaven/client/utility/utility.h"
 
+#include <optional>
```

Comment:
The ordering that I find meaningful for C++ is to include the system includes last, the own library side dependencies first, and the dependent libraries in the middle

The rationale for that is the IWYU thingy (include what you use).  Meaning, if you include your own header and that header fails to include something /it/ needs, the compilation would fail right there

instead of being satisfied by chance by an earlier inclusion of a system dependency
what do you think?